File: 100289/string-muveletek/Stringmuveletek.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.HashSet;
import java.util.Scanner;
import java.util.Set;

public class Stringmuveletek {
  public static void main(String[] args) throws FileNotFoundException {
    megold(args[0]);
  }

  static void megold(String filename) throws FileNotFoundException {
    HashSet<String> egyediSzavak = new HashSet<String>();
    File beolvasandoFajl = new File(filename);
    Scanner scanner = new Scanner(beolvasandoFajl);
    while(scanner.hasNext()) {
      String[] sor = scanner.nextLine().split(" "); // itt lehet, nem szóköz alapján kéne megcsinálni
      String s1 = sor[0];
      String operator = sor[1];
      String s2 = sor[2];
      egyediSzavak.add(osszefuz(s1,operator,s2));
    }
    scanner.close();
    System.out.println(egyediSzavak.size());
  }

  static String osszefuz(String s1, String operator, String s2) {
    if(operator.equals("=>")) {
      return s1.concat(s2);
    }
    return s2.concat(s1);
  }
}

File: 100289/string-muveletek/Stringmuveletek.cs
using System;
using System.IO;
using System.Collections.Generic;
public class Stringmuveletek {
  public static void Main(string[] args) {
    megold(args[0]);
  }

  static void megold(string filename) {
    HashSet<string> egyediSzavak = new HashSet<string>();
    StreamReader beolvasandoFajl = new StreamReader(filename);
    string Line;
    while(!string.IsNullOrEmpty((Line = beolvasandoFajl.ReadLine()))) {
      string[] sor = Line.Split(' ');
      string s1 = sor[0];
      string op = sor[1];
      string s2 = sor[2];
      egyediSzavak.Add(osszefuz(s1,op,s2));
    }
    beolvasandoFajl.Close();
    Console.WriteLine(egyediSzavak.Count);
  }

  static string osszefuz(string s1, string op, string s2) {
    if(op.Equals("=>")) {
      return s1 + s2;
    }
    return s2 + s1;
  }
}